src/IROperator.h

```diff
@@ -820,7 +885,12 @@ inline Expr pow(Expr x, Expr y) {
     if (x.type() == Float(64)) {
         y = cast<double>(y);
         return Internal::Call::make(Float(64), "pow_f64", {x, y}, Internal::Call::Extern);
-    } else {
+    }
+    else if (x.type() == Float(16)) {
+         y = cast<float16_t>(y);
+        return Internal::Call::make(Float(16), "pow_f16", {x, y}, Internal::Call::Extern);
```

Comment:
From a correctness stand point erroring out if the backend doesn't support half (I seem recall @zvookin  mentioning teaching halide to declare what types a target supports) seems like the right thing to do.

However from a convenience stand point if someone has gone to trouble of writing their algorithm half types rewriting for other targets will become a nuisance. Maybe halide should expose some sort of optional mode which allows you to treat ``floatX`` as ``floatY`` for a particular target. E.g. if enabled all float16 expressions will be treated as float32 in the backend. This would also allow you to compile code written using doubles on to targets that don't support doubles without rewriting the algorithm.